src/navigation/Nav.jsx

```diff
@@ -0,0 +1,292 @@
+			},
+			{
+				shrink: true,
+				linkTo: groups.link,
+				label: groups.label,
+				className: `atMedium_display--none ${CLASS_AUTH_ITEM}`,
+				icon: <Icon shape="groups" size="s" className="display--block" />,
+			},
+			{
+				shrink: true,
+				linkTo: messages.link,
+				label: messages.label,
+				className: `navItem--messages ${CLASS_AUTH_ITEM}`,
+				icon: (
+					<Icon
+						shape="messages"
+						size="s"
+						className="display--block atMedium_display--none"
+					/>
+				),
+				hasUpdates: messages.unreadMessages > 0,
+				updatesLabel: updatesLabel,
+			},
+			{
+				shrink: true,
+				linkTo: media.isAtMediumUp ? '' : notifications.link,
+				label: notifications.label,
+				className: cx('navItem--notifications', CLASS_AUTH_ITEM),
+				icon: (
+					<Icon
+						shape="notifications"
+						size="s"
+						className="display--block atMedium_display--none"
+					/>
+				),
+				onClickAction:
+					media.isAtMediumUp && notifications.list.length === 0
+						? notifications.getNotificationsQuery
+						: false,
+				dropdownContent: media.isAtMediumUp && notificationContent,
+				hasUpdates: notifications.unreadNotifications > 0,
+				updatesLabel: updatesLabel,
+			},
+			{
+				shrink: true,
+				linkTo: media.isAtMediumUp ? '' : '/profile/',
+				label: profile.label,
+				labelClassName: 'navItem-label display--block atMedium_display--none',
+				className: cx(CLASS_AUTH_ITEM, 'profileDropdown', {
+					'profileDropdown--hasGroups': Boolean(
+						groups.list && groups.list.length
+					),
+				}),
+				icon: (
+					<Flex noGutters align="center">
+						<FlexItem>
+							<AvatarMember small member={self} />
+						</FlexItem>
+						<FlexItem
+							shrink
+							className="display--none atMedium_display--block"
+						>
+							<Icon
+								shape="chevron-down"
+								size="xxs"
+								className="padding--left-half"
+							/>
+						</FlexItem>
+					</Flex>
+				),
+				onClickAction:
+					media.isAtMediumUp && groups.list.length === 0
+						? profile.getSelfGroupsQuery
+						: false,
+				dropdownContent: media.isAtMediumUp && profileContent,
+			},
+		];
+
+		unauthItems = unauthItems.map((item, i) => {
+			return (
+				<NavItem
+					key={i}
+					shrink={item.shrink}
+					linkTo={item.linkTo}
+					label={item.label}
+					className={item.className}
+					onAction={item.onAction && item.onAction}
+				/>
+			);
+		});
+
+		authItems = authItems.map((item, i) => {
+			return (
+				item && (
+					<NavItem
+						key={i}
+						shrink={item.shrink && item.shrink}
+						linkTo={item.linkTo}
+						label={item.label && item.label}
+						labelClassName={item.labelClassName && item.labelClassName}
+						className={item.className && item.className}
+						icon={item.icon && item.icon}
+						onClickAction={item.onClickAction && item.onClickAction}
+						dropdownContent={item.dropdownContent && item.dropdownContent}
+						hasUpdates={item.hasUpdates && item.hasUpdates}
```

Comment:
i think i commented this before, but do you really need to check for `item.x && item.x`? i think if you just pass `item.x` as the prop and it happens to be null, it's the same thing?